BootloaderCommonPkg/Library/LinuxLib/LinuxLib.c

```diff
@@ -316,11 +330,24 @@ LinuxBoot (
   )
 {
   BOOT_PARAMS   *Bp;
+  UINTN          KernelStart;
 
   Bp = GetLinuxBootParams ();
   if (Bp != NULL) {
     UpdateLinuxBootParams (Bp);
-    JumpToKernel ((VOID *)Bp->Hdr.Code32Start, Bp);
+    KernelStart = (UINTN)Bp->Hdr.Code32Start;
+    if (!IsLongModeEnabled ()) {

```

Comment:
Is it a typo for "!IsLongModeEnabled () " here? from logic it looks should be IsLongModeEnabled ()

Reply:
This means current processor mode. IsLongModeEnabled() will return TRUE if current processor mode is long mode, otherwise return FALSE.
If current processor mode is protected mode, processor supports long mode and Linux has 64-bit entry point, JumpToLongMode will switch to long mode and jump to 64-bit entry point.

Reply:
if IsLongModeEnabled() is TRUE, it would CpuDeadLoop(). Is this the expected behavior?

Reply:
It's TBD since there is no 64-bit code to run at this moment. After enabling Long Mode in previous stage, Long Mode kernel jump code (64-to-64 code, not 32-to-64 jump code) will be added.

Reply:
Got it. That's the confusing part. Ideally IsLongModeEnabled () should be added later when we support 64 bit code.